Geodatenbezug.Test/ProcessingTest.cs

```diff
@@ -79,4 +81,156 @@ public async Task GetTopicsToUpdateTest()
         Assert.AreEqual(BaseTopic.lwb_perimeter_ln_sf, topicsToProcess[1].BaseTopic);
         Assert.AreEqual(Canton.ZG, topicsToProcess[1].Canton);
     }
+
+    [TestMethod]
+    public void GetTokenTest()
+    {
+        var result = new Processing(geodiensteApiMock.Object, loggerMock.Object).GetToken(BaseTopic.lwb_rebbaukataster, Canton.BE);
+        Assert.AreEqual("token2", result);
+    }
+
+    [TestMethod]
+    public void GetTokenFailsTest()
+    {
+        Assert.ThrowsException<KeyNotFoundException>(() => new Processing(geodiensteApiMock.Object, loggerMock.Object).GetToken(BaseTopic.lwb_rebbaukataster, Canton.AI), "Token not found for topic lwb_rebbaukataster and canton AI");
+    }
+
+    [TestMethod]
+    public async Task ProcessTopicTest()
+    {
+        var topic = new Topic
+        {
+            BaseTopic = BaseTopic.lwb_rebbaukataster,
+            TopicName = "lwb_rebbaukataster_v2_0",
+            TopicTitle = "Rebbaukataster",
+            Canton = Canton.AG,
+            UpdatedAt = DateTime.Now.AddHours(-4),
+        };
+        var token = "token1";
+        var processingResult = new ProcessingResult
+        {
+            Code = HttpStatusCode.OK,
+            Reason = "Success",
+            Info = "Processing completed",
+            TopicTitle = topic.TopicTitle,
+            Canton = topic.Canton,
+            DownloadUrl = null,
+        };
+        geodiensteApiMock
+            .Setup(api => api.StartExportAsync(It.IsAny<Topic>(), It.IsAny<string>()))
+            .ReturnsAsync(new HttpResponseMessage(HttpStatusCode.OK));
+        geodiensteApiMock
+            .Setup(api => api.CheckExportStatusAsync(It.IsAny<Topic>(), It.IsAny<string>()))
+            .ReturnsAsync(new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent("{\"status\":\"success\", \"info\":\"Data ready to be downloaded. Provide your credentials to download the data.\", \"download_url\":\"test.com/data.zip\", \"exported_at\":\"2022-03-24T09:31:05.508\"}"), });
+        loggerMock.Setup(LogLevel.Information, $"Verarbeite Thema {topic.TopicTitle} ({topic.Canton})...");
+
+        var result = await new Processing(geodiensteApiMock.Object, loggerMock.Object).ProcessTopic(topic);
+        Assert.AreEqual(processingResult, result);
+        geodiensteApiMock.Verify(api => api.StartExportAsync(It.Is<Topic>(t => t == topic), It.Is<string>(s => s == token)), Times.Once);
```

Comment:
Q: Würde hier auch folgende vereinfachte Schreibweise gehen? 
```suggestion
        geodiensteApiMock.Verify(api => api.StartExportAsync(topic, token), Times.Once);
```